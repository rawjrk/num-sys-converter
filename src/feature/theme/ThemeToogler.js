import styled from "styled-components";
import Button from "./components/Button";
import { BsSun, BsMoonStars } from "react-icons/bs";
import { useSelector, useDispatch } from "react-redux";
import { themeNameSelector, toogleTheme } from "./themeSlice";

const ThemeButton = styled(Button)`
  position: absolute;
  top: 1rem;
  right: 1rem;

  svg {
    font-size: 1.6rem;
    margin: 0.1em;
    margin-bottom: 0;
  }
`;

function ThemeToogler() {
  const themeName = useSelector(themeNameSelector);
  const dispatch = useDispatch();

  return (
    <ThemeButton onClick={() => dispatch(toogleTheme())}>
      {themeName === "darkTheme" ? <BsSun /> : <BsMoonStars />}
    </ThemeButton>
  );
}

export default ThemeToogler;
