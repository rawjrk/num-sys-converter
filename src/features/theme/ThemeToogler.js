import styled from "styled-components";
import { useSelector, useDispatch } from "react-redux";
import PropTypes from "prop-types";
import { themeNameSelector, toogleTheme } from "./themeSlice";
import { Button } from "../shared";

const ThemeButton = styled(Button)`
  position: absolute;
  top: 1rem;
  right: 1rem;

  svg {
    font-size: 1.6rem;
    margin: 0.1em;
    margin-bottom: 0;
  }
`;

function ThemeToogler({ showOnDarkTheme, showOnLightTheme }) {
  const themeName = useSelector(themeNameSelector);
  const dispatch = useDispatch();

  return (
    <ThemeButton onClick={() => dispatch(toogleTheme())}>
      {themeName === "darkTheme" ? showOnDarkTheme : showOnLightTheme}
    </ThemeButton>
  );
}

ThemeToogler.propTypes = {
  showOnDarkTheme: PropTypes.node,
  showOnLightTheme: PropTypes.node,
};

ThemeToogler.defaultProps = {
  showOnDarkTheme: "switch to light theme",
  showOnLightTheme: "switch to dark theme",
};

export default ThemeToogler;
