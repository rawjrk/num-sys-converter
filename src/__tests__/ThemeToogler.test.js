import { screen, fireEvent } from "@testing-library/react";
import ThemeToogler from "../features/theme/ThemeToogler";

const setupOptions = {
  includeTheme: true,
  preloadedState: { theme: { name: "lightTheme" } },
};

const setup = () => {
  return renderWithProviders(<ThemeToogler />, setupOptions);
};

test("renders successfully", () => {
  renderWithProviders(<ThemeToogler />, setupOptions);
  expect(screen.getByRole("button")).toBeInTheDocument();
});

test("renders propper text on init", () => {
  const { store } = setup();

  expect(store.getState().theme.name).toBe("lightTheme");
  expect(screen.queryByText(/to dark/i)).toBeInTheDocument();
  expect(screen.queryByText(/to light/i)).not.toBeInTheDocument();
});

test("renders propper text on switch", () => {
  const { store } = setup();

  expect(store.getState().theme.name).toBe("lightTheme");

  const button = screen.getByRole("button");
  fireEvent.click(button);

  expect(store.getState().theme.name).toBe("darkTheme");
  expect(screen.queryByText(/to light/i)).toBeInTheDocument();
  expect(screen.queryByText(/to dark/i)).not.toBeInTheDocument();
});

test("matches snapshot", () => {
  const tree = snapshotWithProviders(<ThemeToogler />, setupOptions);
  expect(tree).toMatchSnapshot();
});
