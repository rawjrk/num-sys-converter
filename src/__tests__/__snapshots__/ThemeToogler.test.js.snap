// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="Button-sc-cn90ei-0 ThemeToogler__ThemeButton-sc-1e5kdr8-0 jSFdDE KxIsX"
  onClick={[Function]}
>
  switch to dark theme
</button>
`;
